actionpack/lib/action_dispatch/testing/integration.rb

```diff
@@ -648,4 +648,16 @@ def document_root_element
 
     include Behavior
   end
+
+  class IntegrationTest < IntegrationTestCase
+    def initialize(*)
+      ActiveSupport::Deprecation.warn(<<-MSG.squish)
+        ActionDispatch::IntegrationTest has been renamed ActionDispatch::IntegrationTestCase.
+        ActionDispatch::IntegrationTest will be removed in Rails 6.0.
+        Please use ActionDispatch::IntegrationTestCase adapter instead.
+      MSG
+
+      super
```

Comment:
This is great and I'm glad you opened this PR! I think that we should move `IntegrationTestCase` out of `Integration` and into `integration_test_case.rb` and then move the testing directory to `integration_testing` so that those match how system tests work now (top level `SystemTestCase` and directory of dependencies in`system_testing`.